### ISSUE TITLE
Protocol is incorrectly shown as compatible

### ISSUE THREAD

**Issue by rshin** [microsoft/pyright#3445]
**Describe the bug**
A class which does not implement a protocol is treated is compatible with the protocol.

**To Reproduce**
Example code which should raise an error, but does not:
```python
from typing import Generic, TypeVar, Protocol

T = TypeVar("T")

class X(Protocol):
    def f(self) -> int:
        ...


class Y(Generic[T]):
    def f(self) -> T:
        raise NotImplementedError
    
    def g(self) -> X:
        return Y[T]()
```

pyright does not complain, but mypy correctly identifies that `Y[T].f` does not satisfy `X`.

**VS Code extension or command-line**
pyright 1.1.245

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.247, which I just published. It will also be included in the next release of pylance.
